@xen-orchestra/openflow/src/parse-socket.js

```diff
@@ -21,9 +21,11 @@ export default async function* parse(socket) {
     }
 
     // Read the rest of the openflow message
-    data = await readChunk(socket, msgSize - scheme.size)
-    assert.notStrictEqual(data, null)
-    data.copy(buffer, scheme.size, 0, msgSize - scheme.size)
+    if (msgSize > scheme.size) {
+      data = await readChunk(socket, msgSize - scheme.size)
+      assert.notStrictEqual(data, null)
+      data.copy(buffer, scheme.size, 0, msgSize - scheme.size)
+    }
```

Comment:
Idem for changes in this file.

Reply:
other PR merged into this branch, the only file to review is `packages/xo-server-sdn-controller/src/index.js`